### DISCUSSION TITLE
Custom Plugin in library

### DISCUSSION THREAD

**Discussion by guillermo2007** [quartznet/quartznet#1893] [General]
Hi,

I have a question about the reference Plugins in Quartz.

I have some projects with quartz.net and I created a library to register the history of quartz in a database.
I want to reference the dll in other projects and I have a problem. When i reference the project of my principal project, the application runs good, because build works to copy dll´s in destination path, but i don't want this, i want to copy dll manually and don't reference in then main project, only touch appsettings.json with the name of the plugin.

Is it possible?

Best regards

**Comment by guillermo2007** [comment]
@Lahma any idea?

**Comment by lahma** [comment]
Sounds more like MSBuild issue. Just build the library in isolation and copy to destination as some PostBuild event in csproj?

**Comment by guillermo2007** [comment]
I did this and later when i run the application and show this error:

Could not load file or assembly 'Custom.Plugin.History, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File is in destination folder

**Reply by lahma** [reply]
I'm afraid that this still isn't Quartz specific, it's more about runtime dependency handling.

**Reply by guillermo2007** [reply]
Ohhhh, ok.
Thanks Marko, I was waiting for instance the plugin by reflection.

Best regards
